apps/api/src/campaign/campaign.service.ts

```diff
@@ -140,6 +140,31 @@ export class CampaignService {
     return campaign
   }
 
+  async getCampaignByIdWithDefaultVault(campaignId: string): Promise<Campaign> {
+    const campaign = await this.prisma.campaign.findFirst({
+      where: { id: campaignId },
+      include: {
+        campaignFiles: true,
+        vaults: true,
+        incomingTransfers: { select: { amount: true } },
+      },
+    })
+    if (!campaign) {
+      Logger.warn('No campaign record with ID: ' + campaignId)
+      throw new NotFoundException('No campaign record with ID: ' + campaignId)
+    }
+
+
+    if (campaign.vaults.length == 0) {
+      throw new NotFoundException('No vaults found for campaign: ' + campaignId)
+    }
+
+    campaign['defaultVault'] = campaign.vaults[0].id
```

Comment:
When we alter the campaign object we don't actually update the response type of Prisma, so it won't be available during static typing checks.

If we decide to keep this approach we might rename `defaultVault` to `defaultVaultId` to better state the content we keep in.

I'd suggest that we have a helper method like:

```tsx
getDefaultVaultId(campaign: Campaign): string {
  if (campaign.vaults.length == 0) {
    throw new NotFoundException('No vaults found for campaign: ' + campaignId)
  }
  return campaign.vaults[0].id
}
```

Reply:
It turns out we already have: getCampaignVault.
I am using this now.

Reply:
That's even better! Code reuse for the win :top: